### ISSUE TITLE
Question about features and label shape

### ISSUE THREAD

**Issue by HoaiDuyLe** [salesforce/hierarchicalContrastiveLearning#3]
Hi! Thank you for your great work.

I am trying to run your code on the DeepFashion dataset but it seems like my device cannot run. So I would like to ask you what is the shape of the [features] and [labels] in the HMLC class. 
Thank you!

```
class HMLC(nn.Module):
    def __init__(self, temperature=0.07,
                 base_temperature=0.07, layer_penalty=None, loss_type='hmce'):
        super(HMLC, self).__init__()
        self.temperature = temperature
        self.base_temperature = base_temperature
        if not layer_penalty:
            self.layer_penalty = self.pow_2
        else:
            self.layer_penalty = layer_penalty
        self.sup_con_loss = SupConLoss(temperature)
        self.loss_type = loss_type

    def pow_2(self, value):
        return torch.pow(2, value)

    def forward(self, features, labels):
        device = (torch.device('cuda')
                  if features.is_cuda
                  else torch.device('cpu'))
```

**Comment by wyw97** [comment]
Hello, I wonder if you have solved the problem. I want to use this code for my work, so it's the same problem for me. Thanks!
